### PR TITLE
Follow ups resolution mechanism

### PR DESCRIPTION
**Pull request by anvmn**
## Deployment instructions:

```
drush fra -y
drush updb
```

### CODE REVIEW

**Comment by anvmn** [comment]
@nedSaf please review

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 1]
```diff
@@ -0,0 +1,335 @@
+<?php
+
+/**
+ * @file
+ * Drupal needs this blank file.
+ */
+
+/**
+ * Implements hook_node_insert().
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module:18]: Please add a description of what we are doing here.

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    'prenatal_follow_up',
+    'prenatal_encounter',
+  ];
+  if (in_array($node->type, $prenatal_triggers)) {
+    hedley_case_management_handle_prenatal_insert_trigger($node);
+  }
+}
+
+/**
+ * Implements hook_node_update().
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module:47]: Here as well.

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+  }
+}
+
+/**
+ * Returns the bundles for nodes that are considered nutrition follow-ups.
+ *
+ * @return string[]
+ */
+function hedley_case_management_nutrition_follow_up_types() {
+  return [
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]: Can't this be a `const`?

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+/**
+ * Processes create events that trigger acute illness follow-ups resolution.
+ *
+ * @param object $node
+ *   Created node.
+ */
+function hedley_case_management_handle_acute_illness_insert_trigger($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+  // Resolve for which person and illness created node belongs to.
+  if ($node->type == 'acute_illness_encounter') {
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]:
```suggestion
  if ($node->type === 'acute_illness_encounter') {
```

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+ * Processes create events that trigger nutrition follow-ups resolution.
+ *
+ * @param object $node
+ *   Created node.
+ */
+function hedley_case_management_handle_nutrition_insert_trigger($node) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+
+  // Resolve for which person created node belongs to.
+  if ($node->type == 'home_visit_encounter') {
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]:
```suggestion
  if ($node->type === 'home_visit_encounter') {
```

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+  if (in_array($node->type, $prenatal_triggers)) {
+    hedley_case_management_handle_prenatal_insert_trigger($node);
+  }
+}
+
+/**
+ * Implements hook_node_update().
+ */
+function hedley_case_management_node_update($node) {
+  if ($node->type != 'individual_participant') {
```
Comment (nedSaf) [inline server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]:
```suggestion
  if ($node->type !== 'individual_participant') {
```

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module (hunk 4 above)
Comment (nedSaf) [reply server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]: Here and below.

**Review by nedSaf — COMMENTED** [review]
Looks good, minor comments below, also if relevant, some screenshots would be good.

File: server/hedley/modules/custom/hedley_case_management/hedley_case_management.module (hunk 3 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_case_management/hedley_case_management.module]: Can be indeed.

**Pull request merged**